dm/dm/master/scheduler/scheduler_test.go

```diff
@@ -1212,16 +1213,30 @@ func (t *testScheduler) TestTransferSource(c *C) {
 
 	// test fail halfway won't left old worker unbound
 	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/dm/master/scheduler/failToReplaceSourceBound", `return()`), IsNil)
-	c.Assert(s.TransferSource(sourceID1, workerName1), NotNil)
+	c.Assert(s.TransferSource(ctx, sourceID1, workerName1), NotNil)
 	c.Assert(s.bounds[sourceID1], DeepEquals, worker4)
 	c.Assert(worker1.Stage(), Equals, WorkerFree)
 	c.Assert(failpoint.Disable("github.com/pingcap/tiflow/dm/dm/master/scheduler/failToReplaceSourceBound"), IsNil)
 
-	// test can't transfer when there's any running task on the source
+	// set running tasks
 	s.expectSubTaskStages.Store("test", map[string]ha.Stage{sourceID1: {Expect: pb.Stage_Running}})
-	c.Assert(s.TransferSource(sourceID1, workerName1), NotNil)
-	c.Assert(s.bounds[sourceID1], DeepEquals, worker4)
-	c.Assert(worker1.Stage(), Equals, WorkerFree)
+
+	// test can't transfer when running tasks not in sync unit
+	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/dm/master/scheduler/operateQueryWorkerF", `return("notInSyncUnit")`), IsNil)
+	c.Assert(terror.ErrSchedulerRequireRunningTaskInSyncUnit.Equal(s.TransferSource(ctx, sourceID1, workerName1)), IsTrue)
+	c.Assert(failpoint.Disable("github.com/pingcap/tiflow/dm/dm/master/scheduler/operateQueryWorkerF"), IsNil)
+
```

Comment:
...so now we can check the task status is recovered

Reply:
there is real task in this UT actually 😂 but we already test task will auto resume after transfer-source in it [test](https://github.com/Ehco1996/ticdc/blob/a44d9d62b9fcd504348d6a54343498f25ed858af/dm/tests/ha/run.sh#L121)